code/go/0chain.net/smartcontract/dbs/event/process_test.go

```diff
@@ -0,0 +1,42 @@
+package event
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"0chain.net/smartcontract/dbs"
+)
+
+func TestAddEvents(t *testing.T) {
+	access := dbs.DbAccess{
```

Comment:
Maybe we should stop using active databases in unit tests. Instead, use mocket to confirm events lead to the correct SQL querries.

Reply:
This test will be skipped as soon as it realizes there is no database to connect.
We need a better pattern for unit and integration testing which will help us running unit test vs integration test. 

I feel integration testing are equally important
Creating a unit test case using mocket is just like we are testing gorm which we do not intend to do here.
We intend to understand if the query works as per our expectations.

Reply:
Unit test is going to ensure the query is being called, and the code execution goes smooth if Gorm works as expected which is why you might think we are testing gorm, but we actually want to ensure that the queries & in the order which they are being written should be the order they invoked in when running the code. 
    
What do you think, can we also add unit test?

Reply:
Order of the query 🤔
We are just executing one query.

Reply:
@Sriep 
I tried to find there is no place where we are using mocket (Except we created a struct and did not use it).
Also, the mocket documentation did not show how to assert a query. I tried to use the mocket struct in the repo but did not find a way to use it correctly... 

Can you please share with me how to do that?

Because 
>  to test that we send the SQL statement we expect to.

We need to assert the query somewhere but i cannot find that.

Reply:
Yes, we need to start using [mocket](https://github.com/Selvatico/go-mocket/blob/master/DOCUMENTATION.md).

Here is an example I did for blobber https://github.com/0chain/blobber/blob/staging/code/go/0chain.net/blobbercore/handler/object_operation_handler_test.go

There is also sqlmocket, which has more features but does not seem to be fully compatible with gorm https://github.com/DATA-DOG/go-sqlmock/issues/118, https://stackoverflow.com/questions/50164399/sqlmock-with-gorm-insert.